### PR TITLE
added radix sort

### PR DESCRIPTION
**Pull request by sn99**
#3

### CODE REVIEW

**Comment by AnshulMalik** [comment]
Hey @sn99 this is nice work! Can we use the existing `counting_sort` implementation that we have in here? Since it'd be more intuitive to understand.

**Comment by sn99** [comment]
@AnshulMalik I don't quite understand you ?

File: src/sorting/radix_sort.rs [hunk 1]
```diff
@@ -0,0 +1,52 @@
+pub fn radix_sort(data: &mut [isize]) {
+    let mut buckets = vec![vec![]; 256];
+
+    for n in 0..4 {
+        for r in data as &[isize] {
+            let val = *r;
+            let mut which = (val >> (8 * n)) & 0xFFisize;
```
Comment (AnshulMalik) [inline src/sorting/radix_sort.rs]: this part is a bit hard to understand, we can use some comment here or try an easier version :)

**Comment by AnshulMalik** [comment]
Let's have a conversation in Gitter channel :)

**Comment by sn99** [comment]
@AnshulMalik I don't think we can change `radix_sort.rs` to use `counting_sort.rs` functions as we use a completely modified version of *counting sort* for radix sort :)

**Comment by sn99** [comment]
@efugier I don't think sorting algo's work that way, a sorting algo must be complete in itself and simply take a *slice* for sorting

**Comment by efugier** [comment]
They do. What if you want to sort a huge array of 0 and 1, do you really want to walk through your array for nothing?

**Comment by sn99** [comment]
@efugier let's talk on gitter :)

**Comment by efugier** [comment]
You can't pass it negative integers, the type checking wouldn't allow it.

Don't forget that this repo is a compilation of algorithms, and the basic counting sort does not take negative integers nor does it checks for the max and min value. 

A first pass to determine the range of the data would result in a x2 when sorting know ranges, such as bits array, which are the major use cases for this algorithm: very small (known) range but huge array.

So no, the first pass does not belong in here.
Plus, calculating the max is really easy and would just add noise in a more complex thing. 

(Also such discussions belongs here, in public!)

**Comment by AnshulMalik** [comment]
Both of you have got valid points to make here 👍 ,

Let's do a research on this and go back in past to see how it was originally intended to work and we'll implement it that way, 
And we can also use our innovative ideas to improve it in another implementation. But I agree with @efugier that we should have one implementation which is original.

**Comment by efugier** [comment]
I think it's best to revert it :)

As a side note, `my_array.iter().max()` will cleanly and idiomatically give you the maximum if needed !

**Comment by sn99** [comment]
Yeah, I know about `.max()` in applied on iterators but a method that shows how max can be found is better for understanding :)

I have reverted the changes made in `counting_sort.rs` :)

**Pull request closed without merging**